### PR TITLE
feature/application-store added

### PR DESCRIPTION
**Pull request by AshenWijesingha**

### CODE REVIEW

File: src/store/interfaces/IApplication.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  status : string;
+  createdBy: IModifiedBy;
+  updatedBy: IModifiedBy[];
+  deletedBy?: IModifiedBy;
+}
+
+// State Interface
+interface IApplicationState {
+  viewApplication: IApplication | null;
+  viewApplications: IApplication[] | null;
```
[the remark this answers is not in the document]
Comment (AshenWijesingha) [reply src/store/interfaces/IApplication.ts:30]: resolved Ayye

File: src/store/interfaces/IApplication.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  archiveApplication: IApplication | null;
+  updatedApplicationStatus: IApplication | null;
```
[the remark this answers is not in the document]
Comment (AshenWijesingha) [reply src/store/interfaces/IApplication.ts:32]: resolved Ayye

File: src/store/interfaces/IApplication.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  createdBy: IModifiedBy;
+  updatedBy: IModifiedBy[];
+  deletedBy?: IModifiedBy;
+}
+
+// State Interface
+interface IApplicationState {
+  viewApplication: IApplication | null;
+  viewApplications: IApplication[] | null;
+  archiveApplication: IApplication | null;
```
[the remark this answers is not in the document]
Comment (AshenWijesingha) [reply src/store/interfaces/IApplication.ts:31]: resolved Ayye

File: src/store/interfaces/IApplication.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  updatedApplicationStatus: IApplication | null;
+  loading: boolean;
```
[the remark this answers is not in the document]
Comment (AshenWijesingha) [reply src/store/interfaces/IApplication.ts:41]: resolved Ayye

File: src/store/application-store/applicationActionTypes.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+enum ApplicationActionTypes {
```
[the remark this answers is not in the document]
Comment (AshenWijesingha) [reply src/store/application-store/applicationActionTypes.ts:1]: Resolved Ayye

File: src/store/interfaces/IApplication.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import { IModifiedBy } from "./index";
+
+interface IApplication {
```
[the remark this answers is not in the document]
Comment (AshenWijesingha) [reply src/store/interfaces/IApplication.ts:3]: Resolved Ayye

File: src/store/application-store/applicationReducer.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+const initialState: IApplicationState = {
+  viewApplication: null,
+  viewApplications: [],
+  archiveApplication: null,
+  updatedApplicationStatus: null,
+  loading: false,
+  error: null
+};
+
+const applicationReducer = (state = initialState, action: any) => {
```
[the remark this answers is not in the document]
Comment (AshenWijesingha) [reply src/store/application-store/applicationReducer.ts:22]: Resolved Ayye

File: src/store/application-store/applicationActions.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+export const updateApplication = (studentId: string, data: FormData) => {
+  return {
+    type: ApplicationActionTypes.UPDATE_APPLICATION,
+    payload: ApplicationAPI.updateApplication(studentId, data),
+  };
+};
+
+
```
[the remark this answers is not in the document]
Comment (AshenWijesingha) [reply src/store/application-store/applicationActions.ts:53]: Done Ayye

**Review by senuravihanjayadeva — APPROVED** [review]

**Review by rusiruavb — APPROVED** [review]
LGTM

**Pull request merged**